enigma-swing/src/main/java/cuchaz/enigma/gui/panels/IdentifierPanel.java

```diff
@@ -79,6 +80,33 @@ private void onModifierChanged(AccessModifier modifier) {
 	public void refreshReference() {
 		this.deobfEntry = entry == null ? null : gui.getController().project.getMapper().deobfuscate(this.entry);
 
+		// Prevent IdentifierPanel from being rebuilt if you didn't click off.
+		if (this.lastEntry == this.entry && this.nameField != null) {
+			if (!this.nameField.hasChanges()) {
+
+				final String name;
+
+				// Find what to set the name to.
+				if (this.deobfEntry instanceof MethodEntry && ((MethodEntry) this.deobfEntry).isConstructor()) {
+					// Get the parent of the method if it is a constructor.
+					final ClassEntry parent = ((MethodEntry) this.deobfEntry).getParent();
+
+					// Default to <???> if the method has no parent... for some reason.
+					name = parent == null ? "<???>" : parent.isInnerClass() ? parent.getName() : parent.getFullName();
```

Comment:
An `IllegalStateException` should be thrown on `parent == null` imo

Reply:
Fair enough; was written that way to not unexpectedly crash the program if one somehow does occur.
Will change it real quick.